yoti-sdk-api/src/main/java/com/yoti/api/client/spi/remote/ReceiptDetailsFactory.java

```diff
@@ -0,0 +1,107 @@
+package com.yoti.api.client.spi.remote;
+
+import static com.yoti.api.client.spi.remote.call.YotiConstants.RFC3339_PATTERN;
+import static com.yoti.validation.Validation.notNull;
+
+import java.nio.charset.StandardCharsets;
+import java.security.Key;
+import java.security.PrivateKey;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.crypto.spec.SecretKeySpec;
+
+import com.yoti.api.client.ApplicationProfile;
+import com.yoti.api.client.Attribute;
+import com.yoti.api.client.ExtraData;
+import com.yoti.api.client.ExtraDataException;
+import com.yoti.api.client.HumanProfile;
+import com.yoti.api.client.ProfileException;
+import com.yoti.api.client.ReceiptDetails;
+import com.yoti.api.client.spi.remote.call.ShareSessionReceipt;
+import com.yoti.api.client.spi.remote.call.ShareSessionReceiptItemKey;
+import com.yoti.crypto.CipherType;
+import com.yoti.crypto.Crypto;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ReceiptDetailsFactory {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ReceiptDetailsFactory.class);
+
+    private final AttributeListReader attributeListReader;
+    private final ExtraDataReader extraDataReader;
+
+    private ReceiptDetailsFactory(AttributeListReader attributeListReader, ExtraDataReader extraDataReader) {
+        this.attributeListReader = notNull(attributeListReader, "profileReader");
+        this.extraDataReader = notNull(extraDataReader, "extraDataReader");
+    }
+
+    public static ReceiptDetailsFactory newInstance() {
+        return new ReceiptDetailsFactory(
+                AttributeListReader.newInstance(),
+                ExtraDataReader.newInstance()
+        );
+    }
+
+    public ReceiptDetails create(
+            ShareSessionReceipt wrappedReceipt,
+            ShareSessionReceiptItemKey wrappedItemKey,
+            PrivateKey privateKey) throws ProfileException {
+
+        Key receiptKey = decryptReceiptKey(wrappedReceipt.getWrappedKey(), wrappedItemKey, privateKey);
+
+        List<Attribute<?>> profileAttr = attributeListReader.read(wrappedReceipt.getProfile(), receiptKey);
+        List<Attribute<?>> otherPartyProfileAttr = attributeListReader.read(wrappedReceipt.getOtherPartyProfile(), receiptKey);
+        return ReceiptDetails.forReceipt(wrappedReceipt.getId())
+                .sessionId(wrappedReceipt.getSessionId())
+                .profile(new ApplicationProfile(profileAttr))
+                .otherPartyProfile(new HumanProfile(otherPartyProfileAttr))
+                .extraData(parseExtraData(wrappedReceipt.getExtraData(), receiptKey))
+                .rememberMeId(parseRememberMeId(wrappedReceipt.getRememberMeId()))
+                .parentRememberMeId(parseRememberMeId(wrappedReceipt.getParentRememberMeId()))
```

Comment:
As mentioned, all of these _might_ not exist in a successful receipt. I think it would be preferable if this didn't set the values if there's no value to set (cause it wasn't in the receipt representation fetched).